projects/ad4170_iio/app/ad4170_iio.c

```diff
@@ -2022,7 +2300,12 @@ static int32_t ad4170_read_burst_data_spi_dma(uint32_t nb_of_samples,
 		tim8_config();
 	}
 
-	dma_cycle_count = ((nb_of_samples) / rxdma_ndtr) + 1;
+	if (nb_of_samples == rxdma_ndtr) {
+		dma_cycle_count = 1;
+	} else {
+		dma_cycle_count = ((nb_of_samples) / rxdma_ndtr) + 1;
+	}
```

Comment:
I think this can be updated to:
`dma_cyc_cnt = (n_samples + rx_ndtr -1)/rx_ndtr;`

Controls both cases and also one more case missed by the above if-else (when nb_samples is a multiple of rx_dmandtr which gives one extra cycle)